docs/okeps/okep-5233-pre-assigned-pod-net-id.md

```diff
@@ -0,0 +1,175 @@
+setting the `v1.multus-cni.io/default-network` annotation to a singular NetworkSelectionElement 
+object.
+
+To enable using pre-defined MAC and IP addresses on pods attached to a primary UDN,
+the `v1.multus-cni.io/default-network` will be reused, as it is a well-known annotation for 
+configuring the pod's default network. The `k8s.v1.cni.cncf.io/networks` annotation is specific to 
+secondary networks and expects a list of networks, which does not fit well with primary UDNs.
+With the proposed approach, the `k8s.ovn.org/primary-udn-ipamclaim` annotation, used to link a 
+pod with a matching claim, will be deprecated in favor of the `IPAMClaimReference` field in the 
+NetworkSelectionElement.
+It is important that OVN-Kubernetes tries to handle potential MAC and IP address conflicts in a 
+clear and predictable fashion whenever possible.
+
+
+### API Details
+
+TDB
+TODO: Ensure backwards compatibility
+TODO: Block day2 changes through CELs
+
+### Implementation Details
+
+The changes outlined in this enhancement should be configurable. This means a configuration knob
+is required to instruct OVN-Kubernetes on whether to process the annotation described in the 
+[Pod network identity](#pod-network-identity) section.
+
+The `v1.multus-cni.io/default-network` annotation should only be processed for new pods, 
+modifying it after the addresses were allocated won't be reflected in the pods network 
+configuration and ideally this should be blocked through a 
+[Validating Admission Policy](https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/).
+The `NetworkSelectionElement` structure has an extensive list of fields, this enhancement 
+focuses only on the following:
+```cgo
+type NetworkSelectionElement struct {
+	// Name contains the name of the Network object this element selects
+	// TODO: Is the Name needed? What's the convention for default
+	Name string `json:"name"`
+	// IPRequest contains an optional requested IP addresses for this network
+	// attachment
+	IPRequest []string `json:"ips,omitempty"`
+	// MacRequest contains an optional requested MAC address for this
+	// network attachment
+	MacRequest string `json:"mac,omitempty"`
+	// IPAMClaimReference container the IPAMClaim name where the IPs for this
+	// attachment will be located.
+	IPAMClaimReference string `json:"ipam-claim-reference,omitempty"`
+}
+```
+
+Any other field set in the struct will be ignored by OVN-Kubernetes. 
+With `k8s.ovn.org/primary-udn-ipamclaim` being deprecated in favor of the `IPAMClaimReference` field
+in the `NetworkSelectionElement` we have to define the expected behavior. To avoid conflicting 
+settings when `v1.multus-cni.io/default-network` is set the `k8s.ovn.org/primary-udn-ipamclaim` is 
+going to be ignored, it will be reflected in the opposite scenario for backwards compatibility 
+with a plan to remove it in a future release.
+> TODO: How do we deprecate an annotation
```

Comment:
I suggest:
- new "way" (multus default net) is introduced; we read the multus default net annotation; if it exists, that value is used, and the `k8s.ovn.org/primary-udn-ipamclaim` is ignored. Maybe we want to raise to the user when the two are present / conflicted.
- after a couple of releases, we only accept the multus default net way of configuring the primary UDN attachment. The old way - `k8s.ovn.org/primary-udn-ipamclaim` - is rejected always.
- after yet another couple of releases we can drop support for `k8s.ovn.org/primary-udn-ipamclaim` in the code, ignoring it from here on after.

Reply:
I added it to the doc, Other than emitting events I don't see a good way to warn users. I don't think this was being used extensively beyond VM live-migration so I hope that should be enough.